src/main/java/org/example/writer/TransactionWriter.java

```diff
@@ -98,7 +99,7 @@ private void withRetryOnException(DataManipulationTransaction transaction) {
             try {
                 transaction.run();
                 ranSuccessfully = true;
-            } catch (SparkException e) {
```

Comment:
My retry fix was a bit different and I didn't clean up after rebasing, I just rethrew from tryTransaction and let this do the retry

Reply:
Yes yours is a better one. I over looked this function, because I thought that I forgot to add the retry mechanism